src/common/cmd.c

```diff
@@ -206,87 +208,84 @@ static cmdstat_t cmd_exec(const char* line)
     return cmdstat_err;
 }
 
-cmdstat_t cmd_enqueue(const char* text)
+cmdstat_t cmd_enqueue(const char* constText)
 {
-    return cmd_text(text, false);
+    return cmd_text(constText, false);
 }
 
-cmdstat_t cmd_immediate(const char* text)
+cmdstat_t cmd_immediate(const char* constText)
 {
-    return cmd_text(text, true);
+    return cmd_text(constText, true);
 }
 
 static cmdstat_t cmd_text(const char* constText, bool immediate)
 {
-    i32 start, end, i, q;
-    i8 dir;
     if (!constText)
     {
         ASSERT(false);
         return cmdstat_err;
     }
 
     char* text = StrDup(constText, EAlloc_Temp);
+    char** lines = NULL;
+    i32 lineCount = 0;
 
-    i = q = 0;
-    dir = 1;
-
-    if (immediate)
-    {
-        // got to go through lines backwards as we are pushing to the front of the queue.
-        i = q = StrLen(text) - 1;
-        dir = -1;
-
-        // Take care of the line endings at the very end, if any.
-        while (i >= 0 && IsLineEnding(text[i]))
-        {
-            text[i] = 0;
-            i--;
-        }
-    }
-
+    i32 len = 0;
+    i32 q = 0;
 parseline:
-    start = i;
+
+    len = 0;
     q = 0;
-    end = -1;
-    while (i >= 0 && text[i])
+    while (text[len])
     {
-        char c = text[i];
-        i += dir;
-        if (c == '"')
+        char c = text[len];
+        ++len;
+        if (IsQuote(c))
         {
             ++q;
         }
         else if (!(q & 1) && IsLineEnding(c))
         {
-            end = i - dir;
-            text[end] = 0;
+            // line ending becomes null terminator
+            text[len - 1] = 0;
             break;
         }
     }
 
-    // If we're iterating backwards and we reach the front,
-    // that means we didn't zero-out the last line ending, do that now.
-    if (immediate && i < 0 && IsLineEnding(text[start + 1]))
+    if (len > 0)
     {
-        text[start] = 0;
+        char* line = Perm_Calloc(len + 1);
+        memcpy(line, text, len);
+        text += len;
+        len = 0;
+
+        ++lineCount;
+        lines = Temp_Realloc(lines, sizeof(lines[0]) * lineCount);
```

Comment:
Does this resize every time we add to the array? Would it be better to use StrList or something? (Still getting acquainted with your containers--which are very elegant btw)